src/index.mjs

```diff
@@ -65,16 +65,28 @@ export function listen(options) {
 
   const timeoutFn = options.timeoutFn || requestIdleCallback;
 
+  // ray test touch <
+  const prefetchChunks = options.prefetchChunks;
+  // ray test touch >
+
+  // ray test touch <
+  const prefetchHandler = urls => {
+    prefetch(urls, options.priority).then(isDone).catch(err => {
+      isDone(); if (options.onError) options.onError(err);
+    });
+  };
```

Comment:
We should include a PR comment to explain why `prefetchHandler` is needed. Could you add a comment reply below here?  For other contributors, they know `prefetch()` already supports passing an array of URLs but may not be certain why introducing this new `prefetchChunks` option is needed.

Reply:
`prefetchChunks` method is introduced specifically for the purpose of fetching chunk URLs (e.g. /static/js/about.f0141832.chunk.js) not route URLs (e.g. /about) by mapping each route (/about etc) back to the Webpack chunk it corresponded to.

So this method consists of two parts, where:
* one part is for mapping
* the other one for fetching mapped chunk URLs. Fetching is done by `quicklink`'s `prefetch` method.
```
const prefetchChunks = (entry, prefetchHandler) => {
  try {
    const chunkEntry = rmanifest(window._rmanifest_, entry.pathname);
    const chunkURLs = chunkEntry.files.map(file => file.href);
    if (chunkURLs.length) {
      prefetchHandler(chunkURLs);
      return;
    }
  } catch (error) {
    console.log('[prefetchChunks] error => ', error);
  }

  // also prefetch regular links in-viewport
  prefetchHandler(entry.href);
};
```

We pass this `prefetchChunks` method as `quicklink` option so if this option is set then `quicklink` fetches chunks URLs instead of route URLs (of course regular links as mentioned [here](https://github.com/addyosmani/react-quicklink-demo/pull/2#issuecomment-572211523) and @demianrenzulli [reply](https://github.com/addyosmani/react-quicklink-demo/pull/2#issuecomment-572232377) to it)

This `prefetchHandler` is passed as argument to `prefetchChunks` since fetching is ultimately done by `prefetch` method. It's just `prefetch` method with some options and error handling just to avoid code duplication.

The quicklink is designed to fetch regular links(hrefs of anchor tags) in-viewport but when if we pass `prefetchChunks` as option, it fetches chunks by it. If not, the workflow is the same as before.
```
prefetchChunks ? prefetchChunks(entry, prefetchHandler) : prefetchHandler(entry.href);
```

This option is useful for fetching chunks in Single Page Applications like React SPA app.

Reply:
Hmm.. I still don't understand why the separate `prefetchHandler` is needed, sorry 🙈 Can you explain why [this approach](https://github.com/lukeed/route-manifest#usage) doesn't work?

Reply:
Actually it's not needed to use the separate `prefetchHandler`. We can just use that approach.

I just tried to replicate [this](https://github.com/GoogleChromeLabs/quicklink/blob/836f170df4007c05c24b1300e0faf04fb823168b/src/index.mjs#L75)

Shall we update it?

Reply:
@lukeed if you have time to do a second round of reviews, that would be super appreciated. Going to take another look at the latest commits now too.

Reply:
@anton-karlovskiy Looking at https://github.com/lukeed/route-manifest#usage, I feel we may still be missing why we can't achieve the same behavior as `prefetchChunks()` using Quicklink's current `prefetch()` utility. 

Stepping back...

Step 4 in https://github.com/anton-karlovskiy/intersection-observer-react-quicklink-demo#how-to-setup asks developers to use a React HOC in order to get Quicklink working for routes. 

What if `prefetchChunks` was just a part of the HOC code and re-used `prefetch()`?

```js
const prefetchChunks = (entry) => {
  try {
    const chunkEntry = rmanifest(window.__rmanifest, entry.pathname);
    const chunkURLs = chunkEntry.files.map(file => file.href);
    if (chunkURLs.length) {
      prefetch(chunkURLs);
      return;
    }
  } catch (error) {
    console.log(error);
  }
  // also prefetch regular links in-viewport
  prefetch(entry.href);
};
```

That way we would just package https://glitch.com/edit/#!/anton-karlovskiy-intersection-observer-react-quicklink-demo?path=src/lib/quicklink.js:49:0, but could get everything working without expanding Quicklink's API surface.

Would that work? (i.e what we PR in to this repo is the HOC but not land this PR). Or am I missing something? :)

Reply:
@addyosmani cc @lukeed 

Yes, I agree with you. That should be much better option.
I just tried to replicate the usage of options. But no need, come to think of it.

Reply:
@addyosmani cc @lukeed 

Sorry :)
Let me think of it carefully once more and update you.

Reply:
Yeah, I agree. I really don't think there's anything _new_ to be added to `quicklink` here – at least when talking about `prefetch`'s ability to work in its current form. 

Also I'm not even sure that the HoC is needed...but that may just be me.

The "problem" is that the route manifest can have varying shapes. The `route-manifest` package can only traverse the manifest and return a matching `{ files, headers }` object, but _what_ those `files` look like is entirely up to the developer.  By default, `webpack-route-manifest` produces entries that can fulfill `<link rel=preload>` and `Link` HTTP headers – but many projects may only ever care about the URL, for example, and so their manifest entries may skip the object & return URLs only.

If we were to include a HoC, we'd have to cover both use cases (no problem) ***or*** define an interface that requires a `callback` argument, which is used to extract the URL from each of the `files` returned.

This also kinda assumes that `route-manifest` be included as a quicklink dependency. Otherwise including a utility to filter down to URLs doesn't really make sense? Maybe I'm missing something, it's been a while.

Reply:
After I've discussed with @addyosmani, we made a decision to explore the HOC approach and try to address the 90% of use-cases with it. But we would like to review with @luke if he thinks it makes sense once we have something working.

Reply:
Sounds good! Let me know if you need anything 👍

Reply:
Thank you @lukeed.

I'm working on it and update you and @addyosmani.